static/app/components/notificationActions/notificationActionItem.tsx

```diff
@@ -0,0 +1,344 @@
+            <NotificationRecipient>{action.targetDisplay}</NotificationRecipient>
+            <div>{t('service')}</div>
+          </Fragment>
+        );
+      default:
+        // TODO(enterprise): descriptions for email, msteams, sentry_app
+        return null;
+    }
+  };
+
+  const handleDelete = async () => {
+    const endpoint = `/organizations/${organization.slug}/notifications/actions/${action.id}/`;
+    try {
+      await api.requestPromise(endpoint, {
+        method: 'DELETE',
+      });
+      addSuccessMessage(t('Successfully deleted notification action'));
+      onDelete(index);
+    } catch (err) {
+      addErrorMessage(t('Unable to delete notification action'));
+    }
+  };
+
+  const handleCancel = () => {
+    if (action.id) {
+      setIsEditing(false);
+      return;
+    }
+    // Delete the unsaved notification action
+    onDelete(index);
+  };
+
+  const handleSave = async () => {
+    const formProps = getFormProps();
+    addLoadingMessage();
+    // TODO(enterprise): use "requires" to get data to send
+    // This is currently optimized for spike protection
+    const data = cloneDeep(editedAction);
+
+    // Remove keys from the data if they are falsy
+    Object.keys(data).forEach(key => {
+      if (!data[key]) {
+        delete data[key];
+      }
+    });
+
+    try {
+      const resp = await api.requestPromise(formProps.apiEndpoint, {
+        method: formProps.apiMethod,
+        data: {...data, projects: [project.slug]},
+      });
+      addSuccessMessage(formProps.successMessage);
+      onUpdate(index, resp);
+      setEditedAction(resp);
+      setIsEditing(false);
+    } catch (err) {
+      addErrorMessage(formProps.errorMessage);
+    }
+  };
+
+  const handleChange = (name: string, value: any) => {
+    setEditedAction({...editedAction, [name]: value});
+  };
+
+  // Edit button is located outside of the form
+  const renderEditButton = () => {
+    const menuItems: MenuItemProps[] = [
+      {
+        key: 'notificationaction-delete',
+        label: t('Delete'),
+        priority: 'danger',
+        onAction: () => {
+          openConfirmModal({
+            message: t('Are you sure you want to delete this notification action?'),
+            onConfirm: handleDelete,
+          });
+        },
+      },
+    ];
+
+    // No edit mode for Sentry notifications
+    if (serviceType !== NotificationActionService.SENTRY_NOTIFICATION) {
+      menuItems.unshift({
+        key: 'notificationaction-edit',
+        label: t('Edit'),
+        onAction: () => setIsEditing(true),
+      });
+    }
+
+    return (
+      <DropdownMenu
+        items={menuItems}
+        trigger={triggerProps => (
+          <Button
+            {...triggerProps}
+            aria-label={t('Actions')}
+            size="xs"
+            icon={<IconEllipsis direction="down" size="sm" />}
+            data-test-id="edit-dropdown"
+          />
+        )}
+      />
+    );
+  };
+
+  const getFormProps = () => {
```

Comment:
I'm 🤷 on making this its own getter and arg dictionary

there's only 2 paths, so we don't _really_ need to build this intricate argdicts thing.
we could just have an `isEditing` state or something 🤔 
but not gonna block on it. just food for htought